File: embedchain/chunkers/qna_pair.py
from typing import Optional

from langchain.text_splitter import RecursiveCharacterTextSplitter

from embedchain.chunkers.base_chunker import BaseChunker
from embedchain.config.AddConfig import ChunkerConfig

TEXT_SPLITTER_CHUNK_PARAMS = {
    "chunk_size": 300,
    "chunk_overlap": 0,
    "length_function": len,
}


class QnaPairChunker(BaseChunker):
    """Chunker for QnA pair."""

    def __init__(self, config: Optional[ChunkerConfig] = None):
        if config is None:
            config = TEXT_SPLITTER_CHUNK_PARAMS
        text_splitter = RecursiveCharacterTextSplitter(**config)
        super().__init__(text_splitter)

File: embedchain-js/embedchain/chunkers/QnaPair.ts
import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';

import { BaseChunker } from './BaseChunker';

interface TextSplitterChunkParams {
  chunkSize: number;
  chunkOverlap: number;
  keepSeparator: boolean;
}

const TEXT_SPLITTER_CHUNK_PARAMS: TextSplitterChunkParams = {
  chunkSize: 300,
  chunkOverlap: 0,
  keepSeparator: false,
};

class QnaPairChunker extends BaseChunker {
  constructor() {
    const textSplitter = new RecursiveCharacterTextSplitter(
      TEXT_SPLITTER_CHUNK_PARAMS
    );
    super(textSplitter);
  }
}

export { QnaPairChunker };
